app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Chitter < Sinatra::Base
+
+get '/users/sign_up' do
+    erb :"users/sign_up"
+  end
+
+  post '/users/sign_up' do
+    if params[:password] != params[:password_confirmation]
+      flash.now[:errors] = []
+      flash.now[:errors] << "Your passwords didn't match. Try again."
+      erb :"users/sign_up"
+    else
+      user = User.new(name: params[:name], username: params[:username], email: params[:email], password: params[:password])
+      if user.save
+        session[:user_id] = user.id
+        redirect to '/'
+      else
+        flash.now[:errors] = user.errors.full_messages
+        erb :"users/sign_up"
+      end
+    end
+  end
```

Comment:
This controller looks a bit scary! :) I see that you do a validation for password. If you want to do it yourself it should be done in the model, the way data mapper does it. So, data mapper already does this for you... all you need to add in the model is 'validates_confirmation_of: ... '